.github/workflows/build-docker-images.yml

```diff
@@ -30,18 +33,24 @@ jobs:
     - name: Fetch the default branch (develop) for comparison
       run: git fetch origin develop:refs/remotes/origin/develop --depth=1
 
-    - name: Check for Changes in scripts/references
+    - name: Check for Changes in scripts/${{ matrix.module }}
       id: check_changes
       run: |
-        # Check for a merge base, fallback to root commit if none exists
-        if git merge-base --is-ancestor origin/develop HEAD; then
-          DIFF_BASE="origin/develop"
+        # Determine the correct DIFF_BASE
```

Comment:
Added logic to determine changes against develop (when doing a merge) correctly. I tested it with a local merge on develop and used these commands (it seemed to work). So